proto/regen/ecocredit/v1/tx.proto

```diff
@@ -421,3 +427,21 @@ message MsgUpdateProjectMetadata {
 // MsgUpdateProjectMetadataResponse is the Msg/UpdateProjectMetadataResponse
 // response type.
 message MsgUpdateProjectMetadataResponse {}
+
+// MsgBridge is the Msg/Bridge request type.
+message MsgBridge {
+  // MsgCancel is the MsgCancel message type.
+  MsgCancel msg_cancel = 1;
```

Comment:
Looking at this again in more detail, I think it's preferable for us to not nest proto messages that are used as GRPC request messages like this. Ideally when defining msg server API endpoints, each can evolve independently (e.g. `Bridge()` and `Cancel()` should be able to evolve independently). So similar to how we always have explicitly separate `MsgBridgeRepsonse` and `MsgCancelResponse`, we should probably avoid embedding a `MsgCancel` field inside the `MsgBridge` message.

In terms of which fields from `MsgCancel` we need when a user is calling `MsgBridge`, I the following is sufficient:
- holder
- `repeated credits` (could use a similar sub-message to `CancelCredits` called `BridgeCredits`?)

My assumption is that we don't actually need an explicit "reason" in the bridge call, as most of the information we would want to pass in the cancel reason would be describing that it's a bridge action, and possibly what chain its coming from. But all of this info we can just get from the `target` field, and knowing its a `Bridge()` RPC call.

So for filling the "reason" in the cancel event, can we just hardcode it in the message server to set `reason = "bridge"` or `reason = "bridge-${target}"` ?

Reply:
Just to note, this was implemented as specified in the issue. That being said, I do agree with the changes proposed. When filling the reason for `MsgCancel` in the message server implementation, I would be in favor of `"bridge-${target}"`.

Reply:
sorry abt the confusion!